Add materialize-all command

ottomata opened the pull request:
This also makes better guesses at how to sort schema files,
hopefully materializing dependency ones earlier in the loop.

ottomata committed:
Add materialize-all command

This also makes better guesses at how to sort schema files,
hopefully materializing dependency ones earlier in the loop.

Pchelolo commented on index.js:
```diff
@@ -482,6 +487,31 @@ async function findSchemaPaths(schemaBasePath, options = {}) {
+    // (If common is in the title, assume it is likely a dependency schema.)
+    const infoACommon = infoA.title.includes('common');
+    const infoBCommon = infoB.title.includes('common');
+    return infoACommon === infoBCommon ? 0 : (infoACommon ? -1 : 1) ||
```
Perhaps if both have 'common' we can have the one with a shorter path (shorter in terms of number of subdirectories) go first?

ottomata replied:
I can add hierarchy depth too.  Question is what is the better valid hierarchy:  The actual filepath hierarchy, or the title hierarchy?  I use filepath hierarchy above in the materializeModifiedSchemas because I don't parse title at that point.  Perhaps I should?  Title path seems better, as it is more canonical (that's why I use version from $id instead of filename).

ottomata replied:
Either way it would be best to be consistent at when and how I sort before I generate.  Hm.  I will use path hierarchy just to be consistent.

ottomata replied:
Done.

Pchelolo commented:
And it's merge-conflicting now.

ottomata committed:
Use path hierarchy depth as sort critera instead of title compare

This is consistent with the sorting used by materializeModifiedSchemas,
where title is not available.
